src/components/icon/icon.test.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Icon from './icon';
+
+describe('Icon', () => {
+  it('has defaults', () => {
+    const tree = renderer.create((
+      <Icon name={'foobar'} />
+    )).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  describe('Size API', () => {
+    it('renders small, medium, and large sizes', () => {
+      const sizes = ['small', 'medium', 'large'];
+      sizes.forEach((size) => {
+        const tree = renderer.create((
+          <Icon name={'foobar'} size={size} />
+        )).toJSON();
+        expect(tree.props.className).toContain(`size-${size}`);
+      });
+    });
+  });
+
+  describe('focusable API', () => {
+    it('defaults false', () => {
+      const tree = renderer.create((
+        <Icon name={'foobar'} />
+      )).toJSON();
+      expect(tree.props.focusable).toEqual(false);
+    });
+
+    it('sets to true', () => {
+      const tree = renderer.create((
+        <Icon name={'foobar'} focusable={true} />
+      )).toJSON();
+      expect(tree.props.focusable).toEqual(true);
+    });
+  });
+
+  describe('className API', () => {
+    describe('fill', () => {
+      it('sets all fill colors', () => {
+        const colors = ['primary', 'action', 'highlight', 'caution'];
+        colors.forEach((color) => {
+          const tree = renderer.create((
+            <Icon name={'foobar'} fill={color} />
+          )).toJSON();
+          expect(tree.props.className).toContain('icon-base');
+          expect(tree.props.className).toContain(`fill-${color}`);
+          expect(tree.props.className).not.toContain('stroke');
+          expect(tree.props.className).not.toContain('color');
+        });
+      });
+    });
+
+    describe('stroke', () => {
+      it('sets stroke colors turning off fills and color', () => {
+        const colors = ['primary', 'action', 'highlight', 'caution'];
+        colors.forEach((color) => {
+          const tree = renderer.create((
+            <Icon name={'foobar'} stroke={color} />
+          )).toJSON();
+          expect(tree.props.className).toContain('icon-base');
+          expect(tree.props.className).toContain(`stroke-${color}`);
+          expect(tree.props.className).toContain('fill-none');
```

Comment:
Fwiw, stroke based svg icons always turn off the fill and color by default, until you ask for those as well (covered by the combination describe later).